src/plugins/plot/src/chart/MCTChartLineStepAfter.js

```diff
@@ -0,0 +1,58 @@
+/*global define*/
+
+define([
+    './MCTChartSeriesElement'
+], function (
+    MCTChartSeriesElement
+) {
+    'use strict';
+
+    var MCTChartLineStepAfter = MCTChartSeriesElement.extend({
+        removePoint: function (point, index, count) {
+            if (index > 0 && index / 2 < this.count) {
+                this.buffer[index + 1] = this.buffer[index - 1];
+            }
+        },
+        vertexCountForPointAtIndex: function (index) {
+            if (index === 0 && this.count === 0) {
+                return 2;
+            }
+            return 4;
+        },
+        startIndexForPointAtIndex: function (index) {
+            if (index === 0) {
+                return 0;
+            }
+            return 2 + ((index - 1) * 4);
+        },
+        addPoint: function (point, start, count) {
+            if (start === 0 && this.count === 0) {
+                // First point is easy.
+                this.buffer[start] = point.x;
```

Comment:
All the `if..else` have `this.buffer[start] = point.x;` part common. So moving it out of `if...else` would be a bad idea?